### ISSUE TITLE
Failure to deserialize Roslyn AddImportEnabled setting

### ISSUE THREAD

**Issue by mrward** [mono/monodevelop#5179]
VS 7.6.0.1797 (release-7.6).

IDE log error about being unable to deserialize the Roslyn AddImportEnabled setting.

```
Failed to deserialize key: Roslyn\Features\Remote\AddImportEnabled type: System.Boolean value: True
System.Runtime.Serialization.SerializationException: Serialization error.
  at MonoDevelop.Ide.RoslynServices.Options.MonoDevelopGlobalOptionPersister.Deserialize (System.Object value, System.Type optionType) [0x000c3] in /Users/vsts/agent/2.134.2/work/1/s/monodevelop/main/src/core/MonoDevelop.Ide/MonoDevelop.Ide.RoslynServices.Options/MonoDevelopGlobalOptionPersister.cs:237 
  at MonoDevelop.Ide.RoslynServices.Options.MonoDevelopGlobalOptionPersister.TryFetch (Microsoft.CodeAnalysis.Options.OptionKey optionKey, System.Object& value) [0x00040] in /Users/vsts/agent/2.134.2/work/1/s/monodevelop/main/src/core/MonoDevelop.Ide/MonoDevelop.Ide.RoslynServices.Options/MonoDevelopGlobalOptionPersister.cs:104 
```

The ~/Library/Preferences/VisualStudio/7.0/MonoDevelopProperties.xml file has:

```
<Property key="Roslyn\Features\Remote\AddImportEnabled" value="True" />
```

> VS bug [#637709](https://devdiv.visualstudio.com/DevDiv/_workitems/edit/637709)

**Comment by mhutch** [comment]
That's the default value, I wonder how it got serialized...

Side-note: I wonder whether we should have mapped the foo\bar\baz format keys into nested elements...

**Comment by Therzok** [comment]
Bug seems to be caused for having used the 631500 branch.

**Comment by mrward** [comment]
I have not used the master-vsts631500. I mostly use the latest 7.6 release, master 7.7 and less frequently 7.5 stable.

**Comment by Therzok** [comment]
Found the root cause. Will propose a fix.

**Comment by Therzok** [comment]
The bool value is serialized as string, so `Get<object>()` returns a string.

We need to be able to pass the actual type we request, so the PropertyService can handle it.

Adding a new overload for PropertyService.Get<T> with a Type param will help us solve this.

**Comment by xamarin-release-manager** [comment]
> [sync] [VS-19] Comment by **xamarinc**



Fixed in version 8.6.0.802 (master)



Author: therzok  
 Commit: d19a86605a94528f252db1ac0980fcf3d980cd7b (xamarin/monodevelop)
